src/main/java/org/killbill/billing/plugin/avatax/api/AvaTaxTaxCalculator.java

```diff
@@ -203,10 +207,22 @@ private CreateTransactionModel toTaxRequest(final String companyCode,
                                                 final boolean dryRun,
                                                 final boolean shouldCommitDocuments,
                                                 final Iterable<PluginProperty> pluginProperties,
-                                                final LocalDate utcToday) {
-        Preconditions.checkState((originalInvoiceReferenceCode == null && (adjustmentItems == null || adjustmentItems.isEmpty())) ||
-                                 (originalInvoiceReferenceCode != null && (adjustmentItems != null && !adjustmentItems.isEmpty())),
-                                 "Invalid combination of originalInvoiceReferenceCode %s and adjustments %s", originalInvoiceReferenceCode, adjustmentItems);
+                                                final LocalDate utcToday,
+                                                final boolean skipAnomalousAdjustments) {
+        
+        try {
+            Preconditions.checkState((originalInvoiceReferenceCode == null && (adjustmentItems == null || adjustmentItems.isEmpty())) ||
```

Comment:
The original Precondition was a high level sanity check and I understand why in that scenario we would hit it, and therefore why introducing a  lenient mode (`skipAnomalousAdjustments`) avoids hitting such condition. But is this enough?